x-pack/legacy/plugins/task_manager/task_pool.ts

```diff
@@ -73,24 +75,31 @@ export class TaskPool {
     }
   }
 
-  private async attemptToRun(tasks: TaskRunner[]) {
-    for (const task of tasks) {
-      if (this.availableWorkers < task.numWorkers) {
-        return false;
-      }
-
-      if (await task.claimOwnership()) {
-        this.running.add(task);
-        task
-          .run()
-          .catch(err => {
-            this.logger.warn(`Task ${task} failed in attempt to run: ${err.message}`);
-          })
-          .then(() => this.running.delete(task));
-      }
-    }
-
-    return true;
+  private async attemptToRun(tasks: TaskRunner[]): Promise<boolean> {
```

Comment:
My spidey-sense goes all tingly when I see unbounded queues.  But I'm guessing here, that won't be an issue, not completely clear to me.  Even if it isn't an issue, it would be nice to have a debug log message right here, printing the length of the queue.

Reply:
Well, it isn't really an unbounded queue, seeing as the queue is a fresh array every time we pull tasks from ES and the previous queue is GCed when the scope of `attemptToRun` is cleaned up.

I'll think about where we can add debug logging without polluting the generic nature of the `takeWithBackpressure` operator. In theory you might have a dozen of these in memory, if you have different streams being processed, so I'm not sure how best to integrate logging.

But in the case of TM itself logging this stuff, sure, makes sense, but we weren't logging the flow through the `for task in tasks` that was there before, and I don't really see a difference between these two implementations in that regard.

I suspect it might be confusion about the nature of the queue: this isn't a long lived queue in memory, it's just a way of consuming an array in an async manner that takes into account how many items you actually want to `pull` out of it.